tests/unittests/Makefile

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) Microsoft. All rights reserved
+#
+
+include $(CLEAR_VARS)
+
+CLANG := clang++
+
+SRCDIR := Foundation
+SRCFILES := $(wildcard $(SRCDIR)/*.m*)
+RF_SRCFILES := $(wildcard $(SRCDIR)/ReferenceFoundation/*.mm)
+
+OUTDIR := ../../build/Tests/UnitTests/Foundation/OSX
+OUTTARGETS := $(addsuffix .o, $(addprefix $(OUTDIR)/, $(basename $(notdir $(SRCFILES)))))
+OUTTARGETS += $(addsuffix .o, $(addprefix $(OUTDIR)/ReferenceFoundation_, $(basename $(notdir $(RF_SRCFILES)))))
+
+SRC := Foundation/NSArrayTests.mm
+
+INC := ./Foundation/ReferenceFoundation ../frameworks/include ../frameworks/gtest ../frameworks/gtest/include 
+INC += ../frameworks/OSXShims/include Foundation/ReferenceFoundation ../../frameworks/include
+INC_PARAMS := $(INC:%=-I %)
+
+CFLAGS += -std=c++14
+CFLAGS += -x objective-c++
+CFLAGS += $(INC_PARAMS)
+
+Shims : $(OUTDIR)/windows.o $(OUTDIR)/IwMalloc.o
+
+$(OUTDIR)/IwMalloc.o : ../../Frameworks/WinObjCRT/MemoryManagement.cpp
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $<
+
+$(OUTDIR)/windows.o : ../frameworks/OSXShims/src/windows.cpp
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $<
+
+$(OUTDIR)/%.o : Shims $(SRCDIR)/%.m
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $(word 2,$^)
+
+$(OUTDIR)/%.o : Shims $(SRCDIR)/%.mm
+	mkdir -p $(OUTDIR)
+	$(CLANG) -c -o $@ $(CFLAGS) $(word 2,$^) 
```

Comment:
For this and the above, it would be better to move the `Shims` dependency to the `Foundation.UnitTests.exe` target, and then drop the `$(word )` here and just use `$^`